pkg/gardenlet/controller/shoot/care/health.go

```diff
@@ -778,6 +789,31 @@ func CheckNodeAgentLeases(nodeList *corev1.NodeList, leaseList *coordinationv1.L
 	return nil
 }
 
+// CheckForExpiredNodeLeases checks if the number of expired node Leases surpasses 20% of all existing Leases. If yes,
+// an error will be returned. The motivation is that dependency-watchdog is starting to scale down controllers when 60%
+// of the Leases are expired.
+func CheckForExpiredNodeLeases(nodeList *corev1.NodeList, leaseList *coordinationv1.LeaseList, clock clock.Clock) error {
+	nodeNames := sets.Set[string]{}
+	for _, node := range nodeList.Items {
+		nodeNames.Insert(node.Name)
+	}
+
+	var expiredLeases int
+	for _, lease := range leaseList.Items {
+		if lease.Spec.RenewTime.Add(time.Second*time.Duration(*lease.Spec.LeaseDurationSeconds)).Before(clock.Now()) &&
+			// we only care about Leases related to existing nodes
+			nodeNames.Has(lease.Name) {
+			expiredLeases++
+		}
+	}
+
+	if expiredLeasesPercentage := int32(float64(expiredLeases) / float64(len(leaseList.Items)) * 100); expiredLeasesPercentage >= 20 {
```

Comment:
As we are not interested in the exact number would this not be easier to read while yielding basically the same result?

```suggestion
	if expiredLeasesPercentage := 100 * expiredLeases / len(leaseList.Items); expiredLeasesPercentage >= 20 {
```